src/core/utils/global_address.cpp

```diff
@@ -0,0 +1,264 @@
+                }
+
+                memset(address, 0, sizeof(*address));
+                memcpy(&address->mAddress, &config.mPrefix.mPrefix, 8);
+
+                address->mPrefixLength = config.mPrefix.mLength;
+                address->mPreferredLifetime = config.mPreferred ? 0xffffffff : 0;
+                address->mValidLifetime = 0xffffffff;
+
+                if (aIidCreator(address, aContext) != kThreadError_None)
+                {
+                    createRandomIid(address, aContext);
+                }
+
+                otAddUnicastAddress(aInstance, address);
+                break;
+            }
+        }
+    }
+}
+
+ThreadError Slaac::createRandomIid(otNetifAddress *aAddress, void *aContext)
+{
+    (void) aContext;
+
+    for (size_t i = 8; i < sizeof(aAddress[i].mAddress); i++)
+    {
+        aAddress->mAddress.mFields.m8[i] = (uint8_t)otPlatRandomGet();
+    }
+
+    return kThreadError_None;
+}
+
+ThreadError SemanticallyOpaqueIidGenerator::createIid(otNetifAddress *aAddress, void *aContext)
+{
+    ThreadError error = kThreadError_None;
+    otSemanticallyOpaqueIidGeneratorData *data = (otSemanticallyOpaqueIidGeneratorData *)aContext;
+
+    VerifyOrExit(aContext != NULL, error = kThreadError_InvalidArgs);
+
+    for (uint32_t i = 0; i <= 255; i++)
+    {
+        error = createIid(aAddress, data);
+        VerifyOrExit(error == kThreadError_Failed,);
+
+        data->mDadCounter++;
+    }
+
+exit:
+    return error;
+}
+
+ThreadError SemanticallyOpaqueIidGenerator::createIid(otNetifAddress *aAddress,
+                                                      otSemanticallyOpaqueIidGeneratorData *aData)
+{
+    ThreadError error = kThreadError_None;
+    Crypto::Sha256 sha256;
+    uint8_t hash[Crypto::Sha256::kHashSize];
+
+    sha256.Start();
+
+    sha256.Update(aAddress->mAddress.mFields.m8, aAddress->mPrefixLength / 8);
+
+    VerifyOrExit(aData->mInterfaceId != NULL, error = kThreadError_InvalidArgs);
+    sha256.Update(aData->mInterfaceId, aData->mInterfaceIdLength);
+
+    if (aData->mNetworkIdLength)
+    {
+        VerifyOrExit(aData->mNetworkId != NULL, error = kThreadError_InvalidArgs);
+        sha256.Update(aData->mNetworkId, aData->mNetworkIdLength);
+    }
+
+    sha256.Update((uint8_t *) &aData->mDadCounter, sizeof(aData->mDadCounter));
+
+    VerifyOrExit(aData->mSecretKey != NULL, error = kThreadError_InvalidArgs);
+    sha256.Update(aData->mSecretKey, aData->mSecretKeyLength);
+
+    sha256.Finish(hash);
+
+    memcpy(&aAddress->mAddress.mFields.m8[8], &hash[sizeof(hash) - 8], 8);
+
+    VerifyOrExit(!IsAddressRegistered(aAddress, aData->mAddresses), error = kThreadError_Failed);
+    VerifyOrExit(!IsAddressReserved(aAddress), error = kThreadError_Failed);
+
+exit:
+    return error;
+}
+
+bool SemanticallyOpaqueIidGenerator::IsAddressRegistered(otNetifAddress *aCreatedAddress, otNetifAddress *aAddressList)
+{
+    otNetifAddress *address = aAddressList;
+
+    while (address != NULL)
```

Comment:
I meant that application gives pointer returned from otGetUnicastAddresses() to perform this check. I agree Netif::IsUnicastAddress() is better approach. In current implementation SemanticallyOpaqueIidGenerator class has no access to ThreadNetif. I'm not sure how to implement it well. I see following options:
1. Create constructor in SemanticallyOpaqueIidGenerator to hold reference to ThreadNetif. Drawback is increased memory usage even if application does not use RFC 7217.
2. Get reference to ThreadNetif from otInstance. I believe it will be possible in near future. Right now otInstance is empty.
3. Create otIsUnicastAddress function in openthread.cpp and use it in IsAddressRegistered function.
4. Use otGetUnicastAddress in IsAddressRegistered function instead of pointer provided by application.

Which approach should be adopted in this case?